### PR TITLE
build: extend goreleaser to produce multi-arch docker builds

### PR DESCRIPTION
**Pull request by danxmoran**
Hopefully addresses #20057 #18202 #17840 #20416

Had to update goreleaser and rebuild our CI image to add support. Will comment deeper explanations inline.

### CODE REVIEW

File: .goreleaser-nightly.yml [hunk 1]
```diff
@@ -49,18 +49,21 @@ builds:
...
     formats:
       - deb
       - rpm
     bindir: /usr/bin
-    files:
-      "scripts/init.sh": "/usr/lib/influxdb/scripts/init.sh"
-      "scripts/influxdb.service": "/usr/lib/influxdb/scripts/influxdb.service"
-      "scripts/logrotate": "/etc/logrotate.d/influxdb"
-      "scripts/influxdb2-upgrade.sh": "/usr/share/influxdb/influxdb2-upgrade.sh"
+    contents:
```
Comment (danxmoran) [inline .goreleaser-nightly.yml:58]: Collateral damage from the goreleaser version bump, the old form is now deprecated

File: .goreleaser-nightly.yml [hunk 2]
```diff
@@ -108,10 +111,32 @@ dockers:
...
+    image_templates:
+      - "quay.io/influxdb/influxdb-arm64v8:nightly"
+    dockerfile: docker/influxd/Dockerfile
+    extra_files:
+      - docker/influxd/entrypoint.sh
+    build_flag_templates:
+      - "--platform=linux/arm64/v8"
+    use_buildx: true
+
+docker_manifests:
```
Comment (danxmoran) [inline .goreleaser-nightly.yml:135]: I've seen the individual image-builds work, but this section runs after a piece that fails on my laptop from missing credentials. Will need to see what it does on the nightly builds.

File: docker/circleci/Dockerfile [hunk 3]
```diff
@@ -2,34 +2,44 @@ FROM circleci/golang:1.15-node-browsers
...
+# Install musl cross-compiler for ARM64.
+# The archive downloaded below was built using these commands:
+#RUN git clone https://github.com/richfelker/musl-cross-make.git && \
+#    cd musl-cross-make && \
+#    git checkout v${MUSL_CROSS_MAKE_VERSION} && \
+#    make MUSL_VER=${MUSL_VERSION} TARGET=aarch64-unknown-linux-musl install && \
+#    mv output musl-cross && \
+#    tar czf /tmp/musl-cross-aarch64-${MUSL_CROSS_MAKE_VERSION}.tar.gz musl && \
+#    cd .. && \
+#    rm -rf musl-cross-make
```
Comment (danxmoran) [inline docker/circleci/Dockerfile:22]: Building these archives from source takes > 1 hour. Moving the outputs into S3 reduces build time to O(minutes).

File: docker/circleci/Dockerfile [hunk 4]
```diff
@@ -74,8 +90,16 @@ ENV CARGO_TARGET_X86_64_APPLE_DARWIN_LINKER=/usr/local/osxcross/target/bin/x86_6
 
 WORKDIR /go
 
+# Install Docker buildkit to support multi-arch image builds.
+ENV DOCKER_BUILDX_VERSION=0.5.1
+
+RUN mkdir -p /home/circleci/.docker/cli-plugins && \
+    curl -sfL -o /home/circleci/.docker/cli-plugins/docker-buildx \
+        https://github.com/docker/buildx/releases/download/v${DOCKER_BUILDX_VERSION}/buildx-v${DOCKER_BUILDX_VERSION}.linux-amd64 && \
+    chmod a+x /home/circleci/.docker/cli-plugins/docker-buildx
```
Comment (danxmoran) [inline docker/circleci/Dockerfile:99]: The buildx utils used for multi-arch images aren't included in CircleCI's bundled docker client

File: docker/circleci/Dockerfile [hunk 5]
```diff
@@ -74,8 +90,16 @@ ENV CARGO_TARGET_X86_64_APPLE_DARWIN_LINKER=/usr/local/osxcross/target/bin/x86_6
...
+
 # Install goreleaser.
-ENV GORELEASER_VERSION=v0.142.0
+ENV GORELEASER_VERSION=v0.152.0
```
Comment (danxmoran) [inline docker/circleci/Dockerfile:102]: Needed for the `docker_manifests` and `use_buildx` support

File: docker/circleci/Dockerfile [hunk 6]
```diff
@@ -2,34 +2,44 @@ FROM circleci/golang:1.15-node-browsers
...
+# The archive downloaded below was built using these commands:
+#RUN git clone https://github.com/richfelker/musl-cross-make.git && \
+#    cd musl-cross-make && \
+#    git checkout v${MUSL_CROSS_MAKE_VERSION} && \
+#    make MUSL_VER=${MUSL_VERSION} TARGET=aarch64-unknown-linux-musl install && \
+#    mv output musl-cross && \
+#    tar czf /tmp/musl-cross-aarch64-${MUSL_CROSS_MAKE_VERSION}.tar.gz musl && \
+#    cd .. && \
+#    rm -rf musl-cross-make
+RUN curl https://dl.influxdata.com/influxdb-ci/musl/${MUSL_VERSION}/musl-cross/${MUSL_CROSS_MAKE_VERSION}/musl-cross-aarch64-${MUSL_CROSS_MAKE_VERSION}.tar.gz -O && \
```
Comment (lesam) [inline docker/circleci/Dockerfile:23]: Might want to append a date or other uniqueness to the filename so we can know when we last re-generated it. Also might be nice to move this documentation to an md file and link from here.

Comment (danxmoran) [reply docker/circleci/Dockerfile]: Done!

File: docker/circleci/Dockerfile [hunk 7]
```diff
@@ -1,40 +1,16 @@
-FROM circleci/golang:1.15-node-browsers
+FROM cimg/go:1.15.6-browsers
```
Comment (danxmoran) [inline docker/circleci/Dockerfile:1]: The old base images are deprecated in favor of `cimg`: https://circleci.com/docs/2.0/circleci-images/#next-generation-convenience-images

**Review by lesam — APPROVED** [review]

**Pull request merged**